### ISSUE TITLE
Loss - nan (not a number) when training in notebook

### ISSUE THREAD

**Issue by ZacharyForrest** [experiencor/keras-yolo2#237]
Hey there, 

Trying to train on my own dataset in a notebook format. CUDA and CUDNN all present and correct with a TESLA K80 on a GCP virtual machine. 

But! Logging purely nan as the loss when training. Code is pretty much as per the https://github.com/experiencor/basic-yolo-keras/blob/master/Yolo%20Step-by-Step.ipynb, just with my training data

I'm training on one just one class 'truck', and got the non-notebook thing working on that just fine (but slow)

Any thoughts? Thanks for your time, folks 

```
Epoch 1/100
13/13 [==============================] - 50s 4s/step - loss: nan - val_loss: nan

Epoch 00001: val_loss did not improve
Epoch 2/100
13/13 [==============================] - 32s 2s/step - loss: nan - val_loss: nan

Epoch 00002: val_loss did not improve
Epoch 3/100
10/13 [======================>.......] - ETA: 7s - loss: nan 
```

**Comment by experiencor** [comment]
@ZacharyForrest The notebook lacks the warmup stage. You need to change the code in the notebook a bit.

**Comment by txdat** [comment]
i have a question why use same learning rate for both warmup and normal training, bounding boxes become inf?

**Comment by ZacharyForrest** [comment]
Thanks! 

FYI other noobs, i just set WARM_UP_BATCHES to 3 and everything started working

**Comment by shivaram93** [comment]
where and how to use model.save_weights() when training for the first time with our own dataset?

**Comment by rodrigo2019** [comment]
ex.:
```python
yolo = YOLO(backend             = config['model']['backend'],
                input_size          = (config['model']['input_size_h'], config['model']['input_size_w']), 
                labels              = config['model']['labels'], 
                max_box_per_image   = config['model']['max_box_per_image'],
                anchors             = config['model']['anchors'],
                gray_mode           = config['model']['gray_mode'])

yolo.model.save_weights("your_weights.h5")
```

also you can use my [fork](https://github.com/rodrigo2019/keras-yolo2), and save your model in this way:

```python
inference = yolo.get_inference_model()
inference.save("inference_model.h5")
```

It is usefull because it excludes the model dependency from YOLO class, you can use the model just with keras
